File: web/web-frontend/src/app/model/test/event/scenario-start.event.ts
import {RunnerEvent} from "./runner.event";
import {Path} from "../../infrastructure/path/path.model";
import {EventKey} from "./fields/event-key.model";
import {RunnerEventTypeEnum} from "./enums/runner-event-type.enum";
import {Serializable} from "../../infrastructure/serializable.model";
import {Scenario} from "../scenario/scenario.model";
import {StringUtils} from "../../../utils/string-utils.util";
import {ObjectUtil} from "../../../utils/object.util";

export class ScenarioStartEvent implements RunnerEvent, Serializable<ScenarioStartEvent> {
    time: Date;
    eventKey: EventKey;
    eventType: RunnerEventTypeEnum = RunnerEventTypeEnum.SCENARIO_START_EVENT;

    testName: string;
    testFilePath: Path;
    scenario: Scenario;
    scenarioIndex: number;
    tags: Array<string> = [];

    deserialize(input: Object): ScenarioStartEvent {
        this.eventKey = new EventKey().deserialize(input["eventKey"]);
        this.time = new Date(input["time"]);
        this.testName = input["testName"];
        this.testFilePath = Path.deserialize(input["testFilePath"]);
        this.scenario = new Scenario().deserialize(input);
        this.scenarioIndex = ObjectUtil.getAsNumber(input["scenarioIndex"]);
        this.tags = input['tags'] || [];

        return this;
    }

    serialize(): string {
        return undefined;
    }
}

File: old/testerum-backend/testerum-runner/testerum-runner-api/src/main/kotlin/com/testerum/runner/events/model/ScenarioStartEvent.kt
package com.testerum.runner.events.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.testerum.model.infrastructure.path.Path
import com.testerum.model.test.scenario.Scenario
import java.time.LocalDateTime

data class ScenarioStartEvent @JsonCreator constructor(
        @JsonProperty("time")          override val time: LocalDateTime = LocalDateTime.now(),
        @JsonProperty("eventKey")      override val eventKey: String,
        @JsonProperty("testName")      val testName: String,
        @JsonProperty("testFilePath")  val testFilePath: Path,
        @JsonProperty("scenario")      val scenario: Scenario,
        @JsonProperty("scenarioIndex") val scenarioIndex: Int,
        @JsonProperty("tags")          val tags: List<String>
): RunnerEvent
